stl/inc/format

```diff
@@ -371,10 +371,14 @@ public:
 // transition back to state 1 and return true.
 class _GB11_LeftHand_regex {
 private:
-    enum { _Start, _ExtPic } _State = _Start;
+    enum _State_t : uint8_t { _Start, _ExtPic };

```

Comment:
```suggestion
    enum _State_t : bool { _Start, _ExtPic };
```

?

Reply:
Seems reasonable - I don't believe this DFA will need additional states in the future, and if it did, that would already be an ODR concern (i.e. `uint8_t` doesn't buy us any room for expansion).

Reply:
admittedly I did think of just using a bool instead of an enum. I opted for the enum to make the correspondence to the regex in the unicode technical annex explicit